### PR TITLE
feat: a document can be reshared

### PR DESCRIPTION
**Pull request by Ljinod**
With this PR it is possible to share several times the same document. We
detect that a document is reshared when we try to fetch it: we receive a
403 error — the document already exists but we can't access it.
When we receive such error we send the document as is if it was a
creation and the recipient will interpret it as an update.

### CODE REVIEW

File: pkg/couchdb/couchdb.go [hunk 1]
```diff
@@ -214,6 +214,9 @@ func (j JSONDoc) Valid(field, value string) bool {
 		}
 
 		values := strings.Split(value, "/")
+		if len(values) != 2 {
```
Comment (Ljinod) [inline pkg/couchdb/couchdb.go]: This doesn't directly concern the PR, it is to avoid errors due to a wrong formatting of the referenced_by values. I wanted to make a separate PR but I forgot…

File: web/sharings/files.go [hunk 2]
```diff
@@ -345,3 +418,72 @@ func createDirForSharing(fs vfs.VFS, id, name string) error {
...
+			if reflect.DeepEqual(newReference, oldReference) {
+				exists = true
+				break
+			}
+		}
+
+		if !exists {
+			res = append(res, oldReference)
+		}
+	}
```
Comment (nono) [inline web/sharings/files.go]: You can simplify it to:

```go
for _, oldReference := range oldRefs {
	var exists bool
	for _, newReference := range newRefs {
		if reflect.DeepEqual(newReference, oldReference) {
			res = append(res, oldReference)
			break
		}
	}
}
```

File: web/sharings/files.go [hunk 3]
```diff
@@ -345,3 +418,72 @@ func createDirForSharing(fs vfs.VFS, id, name string) error {
...
+			if newTag == oldTag {
+				exists = true
+				break
+			}
+		}
+
+		if !exists {
+			res = append(res, oldTag)
+		}
+	}
```
Comment (nono) [inline web/sharings/files.go]: Same here

File: web/sharings/files.go (hunk 2 above)
Comment (Ljinod) [inline web/sharings/files.go]: Well actually here I don't want to append if there are equal, the aim is to merge without duplicating elements (it does look like a mistake I made in a previous PR).

File: web/sharings/files.go (hunk 3 above)
Comment (Ljinod) [inline web/sharings/files.go]: (Same answer as above :))

File: web/sharings/files.go (hunk 2 above)
Comment (nono) [inline web/sharings/files.go]: My mistake
